### ISSUE TITLE
Implementation of rules is wrong with multiple same letters

### ISSUE THREAD

**Issue by tardypad** [monich/harbour-wordle#1]
Here are a couple examples that show some issues with the coloration of the letters.

![game2](https://user-images.githubusercontent.com/6368493/151510576-869ae24c-3780-4362-8af4-ed60e9008e83.png)

(screenshot used in Harbour)
In the 3rd line, only one E should be colored to match the number of E in the end word

![game1](https://user-images.githubusercontent.com/6368493/151510571-00dda4f9-0cc2-4036-ad52-dc99f96095b8.png)

(from one of my game)
In the 3rd line, the last L shouldn't be colored at all as it's already colored at the right 1st place (and there's no other L in the end word)

For reference, here's a screenshot from the official game that show the validity of my remarks
![official](https://user-images.githubusercontent.com/6368493/151510578-85a04360-f459-4fab-96d8-ff80e2506069.png)

**Comment by monich** [comment]
Indeed, you're right! I'll fix the rules. The JavaScript code of the original game is obfuscated or something, and is hard to read, but the row state evaluation function goes roughly like this (I reformatted it a bit to be more readable):
```js
var e = this.boardState[this.rowIndex],  // row to evaluate
    a = this.solution,                   // the answer
    s = Array(a.length).fill("absent"),  // the state
    t = Array(a.length).fill(true),
    o = Array(a.length).fill(true)

for (var n = 0; n < e.length; n++) {
    if (e[n] === a[n]) {
        s[n] = "correct"
        t[n] = false
        o[n] = false
    }
}

for (var i = 0; i < e.length; i++) {
    var L = e[i]
    if (t[i]) {
	for (var j = 0; j < a.length; j++) {
	    var A = a[j]
	    if (o[j] && L === A) {
		s[i] = "present"
                o[j] = false
		break
	    }
	}
    }
}
```

**Comment by monich** [comment]
Should be fixed by c6a1411a

My example now looks like this:

![whole](https://user-images.githubusercontent.com/5909522/151642280-060077e6-2f40-4c50-bd7c-7575f5bd1a21.png)

your example:

![ledge](https://user-images.githubusercontent.com/5909522/151642299-02660164-16d9-42dd-9d4f-917faedd4dab.png)

and the "official" one:

![banal](https://user-images.githubusercontent.com/5909522/151642319-a264dc68-b6bd-45ad-ba6d-646983a73be7.png)

**Comment by monich** [comment]
Should be fixed in 1.0.1, please give it a try (rpms are available from either [github](https://github.com/monich/harbour-wordle/releases/tag/1.0.1) or [openrepos](https://openrepos.net/content/slava/wordle)). Thanks for reporting it!

**Comment by tardypad** [comment]
Thanks for fixing it so fast :)
It looks fine now from the few examples I had today
